ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/SessionProperties.java

```diff
@@ -36,6 +37,27 @@
   private final boolean internalRequest;
   private final Map<String, String> sessionVariables;
 
+  /**
+   * @param mutableScopedProperties   The streamsProperties of the incoming request
+   * @param ksqlHostInfo              The ksqlHostInfo of the server that handles the request
+   * @param localUrl                  The url of the server that handles the request
+   * @param internalRequest           Flag indicating if request is from within the KSQL cluster
+   * @param sessionVariables          Initial session variables
+   */
+  public SessionProperties(
+      final Map<String, Object> mutableScopedProperties,
+      final KsqlHostInfo ksqlHostInfo,
+      final URL localUrl,
+      final boolean internalRequest,
+      final Map<String, Object> sessionVariables
+  ) {
+    this(mutableScopedProperties, ksqlHostInfo, localUrl, internalRequest);
+    if (sessionVariables != null) {
```

Comment:
Is there ever a valid use case for this to be null? If not, let's use `Objects.requireNonNull()` to validate that. If so, we should consider passing an `Optional` instead to make it explicit.